cmd/sysinfo/sysinfo.go

```diff
@@ -64,68 +80,177 @@ func NewSysinfoCmd() *cobra.Command {
 	flags.BoolVar(&sysinfoSpec.ControllerRoleEnabled, "controller", true, "Include controller-specific sysinfo")
 	flags.BoolVar(&sysinfoSpec.WorkerRoleEnabled, "worker", true, "Include worker-specific sysinfo")
 	flags.StringVar(&sysinfoSpec.DataDir, "data-dir", constant.DataDirDefault, "Data Directory for k0s")
+	flags.StringVarP(&outputFormat, "output", "o", "human", "Output format. Must be one of human|yaml|json")
 
 	return cmd
 }
 
-type cliReporter struct {
-	w      io.Writer
-	colors aurora.Aurora
-	failed bool
+type cliReporter interface {
```

Comment:
What I had in mind:

* Leave the `cliReporter` unmodified.
* Just have the `resultsCollector` to collect the tree.

Have the select in the command look like this:

```go
switch outputFormat {
case "human":
	cli := &cliReporter{
		w:      out,
		colors: aurora.NewAurora(term.IsTerminal(out)),
	}
	if err := probes.Probe(cli); err != nil {
		return err
	}
	if cli.failed {
		return errors.New("sysinfo failed")
	}
	return nil

case "json":
	return collectAndPrint(probes, out, json.Marshal)

case "yaml":
	return collectAndPrint(probes, out, yaml.Marshal)

default:
	return fmt.Errorf("unknown output format: %q", outputFormat)
}
```

And the `collectAndPrint` function could look like this:

```go
func collectAndPrint(probe probes.Probe, out io.Writer, marshal func(any) ([]byte, error)) error {
	var c resultsCollector
	if err := probe.Probe(&c); err != nil {
		return err
	}
	if c.failed {
		return errors.New("sysinfo failed")
	}
	bytes, err := marshal(c.results)
	if err != nil {
		return err
	}

	_, err = out.Write(bytes)
	return err
}
```